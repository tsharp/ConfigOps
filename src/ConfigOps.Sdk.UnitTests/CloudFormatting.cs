using ConfigOps.Sdk.Api.v1alpha1;
using ConfigOps.Sdk.Serialization;
using Snapshooter.Xunit;

namespace ConfigOps.Sdk.UnitTests
{
    public class CloudFormatting
    {
        [Fact]
        public void Validate_AzureCloud_YamlFormat()
        {
            Cloud cloud = new Cloud()
            {
                Metadata =
                {
                    Name = "cloud",
                    Namespace = "microsoft-azure"
                },
                Data =
                {
                    Provider = "AzureCloud",
                    ExtensionData =
                    {
                        { "subscriptionId", "12345678-1234-1234-1234-123456789012" },
                        { "tenantId", "12345678-1234-1234-1234-123456789012" },
                        { "clientId", "12345678-1234-1234-1234-123456789012" },
                        { "clientSecret", "12345678-1234-1234-1234-123456789012" }
                    }
                }
            };

            cloud
                .ObjectToYaml()
                .MatchSnapshot();
        }

        [Fact]
        public void Validate_AwsCloud_YamlFormat()
        {
            Cloud cloud = new Cloud()
            {
                Metadata =
                {
                    Name = "cloud",
                    Namespace = "amazon-aws"

                },
                Data =
                {
                    Provider = "AWS",
                    ExtensionData =
                    {
                        { "subscriptionId", "12345678-1234-1234-1234-123456789012" },
                        { "tenantId", "12345678-1234-1234-1234-123456789012" },
                        { "clientId", "12345678-1234-1234-1234-123456789012" },
                        { "clientSecret", "12345678-1234-1234-1234-123456789012" }
                    }
                }
            };

            cloud
                .ObjectToYaml()
                .MatchSnapshot();
        }
    }
}
