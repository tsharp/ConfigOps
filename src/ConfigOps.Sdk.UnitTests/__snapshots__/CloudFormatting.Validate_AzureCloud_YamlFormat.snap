data:
  subscriptionId: 12345678-1234-1234-1234-123456789012
  tenantId: 12345678-1234-1234-1234-123456789012
  clientId: 12345678-1234-1234-1234-123456789012
  clientSecret: 12345678-1234-1234-1234-123456789012
apiVersion: config.kuipersys.com/v1alpha1
kind: Cloud
metadata:
  name: cloud
  namespace: microsoft-azure
  inheritedConfigs:
